### PR TITLE
Module definition and wrapping for convoy

### PR DESCRIPTION
**Pull request by hatched**
This modifies the module dependencies and wrappings so that convoy can generate a proper modules list.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2585/
Test FAILed.

File: app/assets/javascripts/FileSaver.js [hunk 1]
```diff
@@ -12,236 +12,244 @@
...
-));
-// `self` is undefined in Firefox for Android content script context
-// while `this` is nsIContentFrameMessageManager
-// with an attribute `content` that corresponds to the window
+	if (typeof module !== "undefined") module.exports = saveAs;
 
-if (typeof module !== "undefined") module.exports = saveAs;
+}, '', {
+	requires: []
+});
```
Comment (jaycee) [inline app/assets/javascripts/FileSaver.js]: This file became virtually unreadable; safe to say the difference is being wrapped in the YUI.add stuff?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2587/
Test PASSed.

File: app/subapps/browser/views/charm.js [hunk 2]
```diff
@@ -374,7 +374,12 @@ YUI.add('subapp-browser-charmview', function(Y) {
...
     'juju-models',
     'juju-templates',
-    'juju-views',
+    'container-token',
+    'help-dropdown',
+    'user-dropdown',
+    'machine-token',
+    'juju-serviceunit-token',
+    'juju-view-utils',
     'juju-view-utils',
```
Comment (jaycee) [inline app/subapps/browser/views/charm.js]: So now we have juju-view-utils twice?

File: app/assets/javascripts/FileSaver.js (hunk 1 above)
Comment (jaycee) [inline app/assets/javascripts/FileSaver.js]: @hatched confirmed on IRC that this is the case.

**Comment by jaycee** [comment]
Aside from the duplication of juju-view-utils in that one block, this looks good. :+1: 

QA to follow.

**Comment by jaycee** [comment]
In both prod and debug I was able to deploy services, work with machine view, work with config, etc.

QA OK.

**Comment by mitechie** [comment]
:+1:

**Comment by hatched** [comment]
Thanks for the reviews! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**